src/App.tsx

```diff
@@ -1,93 +1,81 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { useLocalStorage } from './hooks/useLocalStorage';
+import { TodoListContext } from './context/TodoListContext';
+import { TodoStatus } from './types/TodoStatus';
 
 export const App: React.FC = () => {
-  return (
-    <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
-
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [filterBy, setFilterBy] = useState<TodoStatus>(TodoStatus.All);
 
-        <ul className="todo-list" data-cy="todoList">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const visibleTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterBy) {
+        case TodoStatus.Active:
+          return !todo.completed;
+        case TodoStatus.Completed:
+          return todo.completed;
+        case TodoStatus.All:
+          return todo;
+        default:
+          return true;
```

Comment:
redundant to filter todos in case TodoStatus.All. 
filter inside cases TodoStatus.Active and TosoStatus.completed and return array of filtered todos.
by default return todos.